### PR TITLE
(BIDS-2175) fix the pagination and Last page

### PR DESCRIPTION
**Pull request by KostasGiannakakis**
### <samp>🤖 Generated by Copilot at 6701c09</samp>

Fixed a bug in `slots.html` that prevented the search filter from applying correctly. Used the DataTables API to set the search value after the table initialization.

### CODE REVIEW

File: templates/slots.html [hunk 1]
```diff
@@ -60,10 +60,11 @@
       }
       var usp = new URLSearchParams(window.location.search)
       var q = usp.get("q")
+      var t = $("#slots").DataTable(tblOpts)
```
Comment (D13ce) [inline templates/slots.html:63]: **Minor:** This should be placed within the if below.

**Review by D13ce — COMMENTED** [review]
Changes look good, I have left a minor suggestion though. ptal

File: templates/slots.html (hunk 1 above)
Comment (KostasGiannakakis) [reply templates/slots.html:63]: well, not really. This line actually creates the Datatable the first time. The initial draw. Then we check if we had anything as a query parameter and if yes, we redraw the table with the `t.draw()` using the new search parameter(s)

**Review by D13ce — APPROVED** [review]
lgtm

**Pull request merged**